salt/modules/x509_v2.py

```diff
@@ -0,0 +1,1933 @@
+
+            ``URI:http://example.com/myca.crl, URI:http://example.org/my.crl``
+
+            There is also support for more attributes using the full form:
+
+            .. code-block:: yaml
+
+                - crlDistributionPoints:
+                    - fullname: URI:http://example.com/myca.crl
+                      crlissuer: DNS:example.org
+                      reasons:
+                        - keyCompromise
+                    - URI:http://example.org/my.crl
+
+        certificatePolicies
+            ``critical, 1.2.4.5, 1.1.3.4``
+
+            Again, there is support for more attributes using the full form:
+
+            .. code-block:: yaml
+
+                - certificatePolicies:
+                    critical: true
+                    1.2.3.4.5: https://my.ca.com/pratice_statement
+                    1.2.4.5.6:
+                      - https://my.ca.com/pratice_statement
+                      - organization: myorg
+                        noticeNumbers: [1, 2, 3]
+                        text: mytext
+
+        policyConstraints
+            ``requireExplicitPolicy:3,inhibitPolicyMapping:1``
+
+        inhibitAnyPolicy
+            The value is just an integer: ``- inhibitAnyPolicy: 1``
+
+        nameConstraints
+            ``critical,permitted;IP:192.168.0.0/255.255.0.0,permitted;email:.example.com,excluded;email:.com``
+
+            .. code-block:: yaml
+
+                - nameConstraints:
+                    critical: true
+                    permitted:
+                      - IP:192.168.0.0/24
+                      - email:.example.com
+                    excluded:
+                      - email:.com
+        noCheck
+            This extension does not take any values, except ``critical``. Just the presence
+            in the keyword args will include it.
+
+        tlsfeature
+            ``status_request``
+
+        For more information, visit the `OpenSSL docs <https://www.openssl.org/docs/man3.0/man5/x509v3_config.html>`_.
+    """
+    if encoding not in ["der", "pem", "pkcs7_der", "pkcs7_pem", "pkcs12"]:
+        raise CommandExecutionError(
+            f"Invalid value '{encoding}' for encoding. Valid: "
+            "der, pem, pkcs7_der, pkcs7_pem, pkcs12"
+        )
+    if kwargs.get("digest", "sha256").lower() not in [
+        "sha1",
+        "sha224",
+        "sha256",
+        "sha384",
+        "sha512",
+        "sha512_224",
+        "sha512_256",
+        "sha3_224",
+        "sha3_256",
+        "sha3_384",
+        "sha3_512",
+    ]:
+        raise CommandExecutionError(
+            f"Invalid value '{kwargs['digest']}' for digest. Valid: sha1, sha224, "
+            "sha256, sha384, sha512, sha512_224, sha512_256, sha3_224, sha3_256, "
+            "sha3_384, sha3_512"
+        )
+    if "der" == encoding and append_certs:
+        raise SaltInvocationError("Cannot encode a certificate chain in DER")
+    if "pkcs12" == encoding and "private_key" not in kwargs:
```

Comment:
```suggestion
    if encoding == "pkcs12" and "private_key" not in kwargs:
```
Checking something pass with what to expect(this is more readable)